core/domain/blog_domain.py

```diff
@@ -287,25 +339,27 @@ def update_url_fragment(self, new_url_fragment):
         self.require_valid_url_fragment(new_url_fragment)
         self.url_fragment = new_url_fragment
 
-    def update_thumbnail_filename(self, new_thumbnail_filename):
+    def update_thumbnail_filename(
+        self, new_thumbnail_filename: Optional[str]
+    ) -> None:
         """Updates the thumbnail filename of a blog post object.
 
         Args:
             new_thumbnail_filename: str|None. The updated thumbnail filename
                 for the blog post.
         """
-        self.require_valid_thumbnail_filename(self.thumbnail_filename)
+        self.require_valid_thumbnail_filename(self.thumbnail_filename) # type: ignore[arg-type]
```

Comment:
Why is the ignore added?

Reply:
`self.thumbnail_filename` is Optional[str] but 1st argument of `require_valid_thumbnail_filename` is supposed to be `str`.

Reply:
Should I add a validation method here, So that I can remove type ignore?

Reply:
Btw one more doubt, we are assigning `self.thumbnail_filename` with `new_thumbnail_filename`. In that case shouldn't we be validating `new_thumbnail_filename` instead of `self.thumbnail_filename`

Reply:
Good point, yeah we should! Please do both then.

Reply:
I guess no need of validating `self.thumbnail_filename` because it is only getting assigned and its value isn't being used.

Reply:
What I meant is to do 
```python
if isinstance(new_thumbnail_filename, str):
    self.require_valid_thumbnail_filename(new_thumbnail_filename)
```

Reply:
This one also seems a bit incorrect.
Cause suppose what we have for `new_thumbnail_filename` is `123` or something other than `str` or `None`.
If this is what I do it won't validate the case where `new_thumbnail_filename` is some other datatype.
So shouldn't it be what I did now?

Reply:
```python3
if isinstance(new_thumbnail_filename, str) and new_thumbnail_filename:
    self.require_valid_thumbnail_filename(new_thumbnail_filename)
```
this one is the current implementation
so if it is None, it directly avoids the validation and it validates other cases as well.